rust/candid/src/de.rs

```diff
@@ -548,7 +548,8 @@ impl<'de, 'a> de::Deserializer<'de> for &'a mut Deserializer<'de> {
     {
         self.unroll_type()?;
         check!(
-            *self.expect_type == TypeInner::Null && *self.wire_type == TypeInner::Null,
+            *self.expect_type == TypeInner::Null
+                && matches!(*self.wire_type, TypeInner::Null | TypeInner::Reserved),
```

Comment:
So this was a different bug, not accepting `Reserved`?

Reply:
It's the same. When skipping null, we assign the wire type to be `reserved`.